sdk/nodejs/resource.ts

```diff
@@ -265,6 +265,14 @@ export abstract class Resource {
     // eslint-disable-next-line @typescript-eslint/naming-convention,no-underscore-dangle,id-blacklist,id-match
     readonly __pluginDownloadURL?: string;
 
+    /**
+     * Private field containing the type ID for this object. Useful for implementing `isInstance` on
+     * classes that inherit from `Resource`.
+     * @internal
+     */
+    // eslint-disable-next-line @typescript-eslint/naming-convention,no-underscore-dangle,id-blacklist,id-match
+    public readonly __pulumiType: string;
```

Comment:
Why did this have to move?

Reply:
Because Component resources implement isInstance as well but they didn't have __pulumiType set on them. So we either duplicated the logic from CustomResource to ComponentResource, or just moved the logic down into Resource.